### PR TITLE
build: gitian, linting and macOS-specific QoL improvements (part 1)

### PR DESCRIPTION
**Pull request by kittywhiskers**
This is a prerequisite for C++17 support on macOS.

Contains patches https://github.com/UdjinM6/dash/commit/03d67bcc7d484bb4e405325aa609240bbd5cc97b ... https://github.com/UdjinM6/dash/commit/3f7d69352502b2fda758e5684e98117279b67fbe 

Part 1: #4255 (this pull request)
Part 2: #4378
Part 3: #4379

### CODE REVIEW

**Comment by UdjinM6** [comment]
Note: 5041 was already backported earlier. It was superseded by 12783 later that's why the code from 5041 is "missing".

File: contrib/gitian-build.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def verify():
 def main():
     global args, workdir
 
-    parser = argparse.ArgumentParser(usage='%(prog)s [options] signer version')
+    parser = argparse.ArgumentParser(description='Script for running full Gitian builds.')
```
Comment (PastaPastaPasta) [inline contrib/gitian-build.py:138]: 15236 a force merge is not appropriate here. Please find what other backports may be needed.

File: depends/hosts/darwin.mk [hunk 2]
```diff
@@ -1,9 +1,8 @@
 OSX_MIN_VERSION=10.12
-OSX_SDK_VERSION=10.11
```
Comment (PastaPastaPasta) [inline depends/hosts/darwin.mk:2]: 16392 needs changes in macdeploy/README.md

File: depends/patches/qt/mac-qmake.conf [hunk 3]
```diff
@@ -18,7 +18,7 @@ QMAKE_APPLE_DEVICE_ARCHS=x86_64
 !host_build: QMAKE_CFLAGS += -target $${MAC_TARGET}
 !host_build: QMAKE_OBJECTIVE_CFLAGS += $$QMAKE_CFLAGS
 !host_build: QMAKE_CXXFLAGS += $$QMAKE_CFLAGS
-!host_build: QMAKE_LFLAGS += -target $${MAC_TARGET} -mlinker-version=$${MAC_LD64_VERSION}
+!host_build: QMAKE_LFLAGS += -target $${MAC_TARGET}
```
Comment (PastaPastaPasta) [inline depends/patches/qt/mac-qmake.conf:21]: 16392 missing changes in build-osx.md

File: doc/release-notes.md [hunk 4]
```diff
@@ -19,13 +19,27 @@ How to Upgrade
...
-dashd/dash-qt (on Linux). If you upgrade after DIP0003 activation and you were
-using version < 0.13 you will have to reindex (start with -reindex-chainstate
+shut down (which might take a few minutes in some cases), then run the
+installer (on Windows) or just copy over `/Applications/Dash-Qt` (on Mac)
+or `dashd`/`dash-qt` (on Linux).
+
+Upgrading directly from a version of Dash Core that has reached its EOL is
+possible, but it might take some time if the data directory needs to be migrated. Old
+wallet versions of Dash Core are generally supported. If you upgrade after DIP0003 activation
+and you were using version < 0.13 you will have to reindex (start with -reindex-chainstate
```
Comment (PastaPastaPasta) [inline doc/release-notes.md:29]: 20419 this should be in a different PR if done

File: depends/Makefile [hunk 5]
```diff
@@ -127,10 +138,10 @@ $(host_prefix)/share/config.site : config.site.in $(host_prefix)/.stamp_$(final_
...
             -e 's|@CC@|$(toolchain_path)$(host_CC)|' \
             -e 's|@CXX@|$(toolchain_path)$(host_CXX)|' \
-            -e 's|@AR@|$(toolchain_path)$(host_AR)|' \
-            -e 's|@RANLIB@|$(toolchain_path)$(host_RANLIB)|' \
-            -e 's|@NM@|$(toolchain_path)$(host_NM)|' \
-            -e 's|@STRIP@|$(toolchain_path)$(host_STRIP)|' \
+            -e 's|@AR@|$(binutils_path)$(host_AR)|' \
+            -e 's|@RANLIB@|$(binutils_path)$(host_RANLIB)|' \
+            -e 's|@NM@|$(binutils_path)$(host_NM)|' \
+            -e 's|@STRIP@|$(binutils_path)$(host_STRIP)|' \
```
Comment (PastaPastaPasta) [inline depends/Makefile]: 17919 missing changes in depends/README.md

File: contrib/gitian-descriptors/gitian-linux.yml [hunk 6]
```diff
@@ -50,7 +50,7 @@ script: |
   export GZIP="-9n"
   export TAR_OPTIONS="--mtime="$REFERENCE_DATE\\\ $REFERENCE_TIME""
   export TZ="UTC"
-  export BUILD_DIR=`pwd`
+  export BUILD_DIR="$PWD"
```
Comment (PastaPastaPasta) [inline contrib/gitian-descriptors/gitian-linux.yml:53]: 17361 missing changes in lint_04_install.sh

File: depends/config.site.in [hunk 7]
```diff
@@ -1,4 +1,14 @@
-depends_prefix="`dirname ${ac_site_file}`/.."
+# shellcheck shell=sh disable=SC2034 # Many variables set will be used in
```
Comment (PastaPastaPasta) [inline depends/config.site.in:1]: 20359 doesn't seem partial

File: depends/patches/qt/freetype_back_compat.patch [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
++++ b/qtbase/src/platformsupport/fontdatabases/freetype/qfontengine_ft.cpp
+@@ -898,7 +898,7 @@ bool QFontEngineFT::init(FaceId faceId, bool antialias, GlyphFormat format,
+         }
+     }
+ #if defined(FT_FONT_FORMATS_H)
+-    const char *fmt = FT_Get_Font_Format(face);
++    const char *fmt = FT_Get_X11_Font_Format(face);
+     if (fmt && qstrncmp(fmt, "CFF", 4) == 0) {
+         FT_Bool no_stem_darkening = true;
+         FT_Error err = FT_Property_Get(qt_getFreetype(), "cff", "no-stem-darkening", &no_stem_darkening);
```
Comment (PastaPastaPasta) [inline depends/patches/qt/freetype_back_compat.patch:28]: 19761 why not add depends/patches/zeromq/remove_libstd_link.patch ?

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: test/functional/feature_addressindex.py [hunk 9]
```diff
@@ -136,7 +136,7 @@ def run_test(self):
         tx.vout = [CTxOut(10, scriptPubKey), CTxOut(11, scriptPubKey)]
         tx.rehash()
 
-        signed_tx = self.nodes[0].signrawtransactionwithwallet(binascii.hexlify(tx.serialize()).decode("utf-8"))
+        signed_tx = self.nodes[0].signrawtransactionwithwallet(tx.serialize().hex())
```
[the remark this answers is not in the document]
Comment (UdjinM6) [reply test/functional/feature_addressindex.py:139]: upstream has no `-addressindex`/`feature_addressindex.py`

File: doc/release-notes.md (hunk 4 above)
Comment (kittywhiskers) [reply doc/release-notes.md:29]: Can't make it a different PR, minimum SDK version is a prerequisite. Will move documentation update to a new PR. Marking as partial

File: depends/config.site.in (hunk 7 above)
Comment (kittywhiskers) [reply depends/config.site.in:1]: It's not, was marked partial when debugging `PYTHONPATH` errors, then reverted without reverting back message, resolved

File: .travis.yml [hunk 10]
```diff
@@ -294,7 +294,7 @@ after_success:
     - stage: test
       env: >-
         HOST=x86_64-apple-darwin14
-        PACKAGES="cmake imagemagick libcap-dev librsvg2-bin libz-dev libbz2-dev libtiff-tools python3-dev python3-setuptools-git"
+        PACKAGES="cmake imagemagick libcap-dev librsvg2-bin libz-dev libbz2-dev libtiff-tools python3-dev python3-setuptools"
```
[the remark this answers is not in the document]
Comment (kittywhiskers) [reply .travis.yml:297]: Resolved

File: depends/README.md [hunk 11]
```diff
@@ -80,4 +80,3 @@ Additional targets:
 
 - [description.md](description.md): General description of the depends system
 - [packages.md](packages.md): Steps for adding packages
-
```
[the remark this answers is not in the document]
Comment (kittywhiskers) [reply depends/README.md]: Fragment caused by buggy editor in VM, resolved

File: depends/Makefile (hunk 5 above)
Comment (kittywhiskers) [reply depends/Makefile]: Resolved

File: depends/patches/qt/mac-qmake.conf (hunk 3 above)
Comment (kittywhiskers) [reply depends/patches/qt/mac-qmake.conf:21]: Resolved

File: depends/hosts/darwin.mk (hunk 2 above)
Comment (kittywhiskers) [reply depends/hosts/darwin.mk:2]: Resolved

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see https://github.com/UdjinM6/dash/commit/5658aea8eda482316183626dbf20ab66dc6d4c1f + pls squash 17550 and 15236 pairs so that we would have 1 commit per backported PR

**Comment by UdjinM6** [comment]
FYI: Adding PR 16392 (708cc7575d4d73f465304cd30ad0c4b46b266810 from #4378 + "more" https://github.com/UdjinM6/dash/commit/efa6796c1ba1a3d99bdc820e19d7e025a16e3a61) fixes macos build in CI, see https://gitlab.com/UdjinM6/dash/-/pipelines/362805572

**Review by UdjinM6 — APPROVED** [review]
LGTM :+1:

utACK

**Comment by UdjinM6** [comment]
ping @PastaPastaPasta

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
Noticed few minor issues + should rename 15236 to just "merge ..." now (not "force-merge ...")

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit
